middleware/domain/source/discovery/state.cpp

```diff
@@ -228,19 +228,115 @@ namespace casual
 
          bool Accumulate::bypass() const noexcept
          {
-            Trace trace{ "domain::discovery::state::Accumulate::bypass"};
-
-
             // if we have an ongoing deadline we keep accumulation until the deadline kicks in.
             if( m_heuristic.deadline)
                return false;
 
-            log::line( verbose::log, "bypass-heuristic: ", m_heuristic);
-
             // bypass if in-flight is within the "window", the lowest/first load level
             return m_heuristic.in_flight() <= m_heuristic.in_flight_window;
          }
 
+         namespace in::flight
+         {
+            namespace content
+            {
+               namespace cache
+               {
+                  void Mapping::add( const common::strong::correlation::id& correlation, const std::vector< std::string>& resources)
+                  {
+                     for( auto& resource : resources)
+                     {
+                        m_correlation_to_resource.emplace( correlation, resource);
+                        ++m_resource_count[ resource];
+                     }
+                  }
+
+                  std::vector< std::string> Mapping::extract( const common::strong::correlation::id& correlation)
+                  {
+                     std::vector< std::string> result;
+
+                     auto make_range = []( auto pair) { return range::make( pair.first, pair.second);};
+
+                     auto resources = make_range( m_correlation_to_resource.equal_range( correlation));
+
+                     result.reserve( resources.size());
+
+                     for( auto& pair : resources)
+                     {
+                        // clean upp total "cached" services
+                        if( auto found = algorithm::find( m_resource_count, pair.second); --found->second == 0)
```

Comment:
Personally I think this type of compound statement is somewhat hard to read and grasp. But that is just because I am not used to it. :-)

Reply:
You mean with _if init statement_ in general?

Reply:
In some cases you can narrow the scope of a variable to the if scope with _init-statement_, which often is worth it. In this particular case it could be done with two nested if statements. Which is ok if it's one-liners, I think.

Reply:
If there is a very long init statement the actual condition appear at the very far right. 
Perhaps a different format is clearer. I.e 
if( init...;
     condition)
   statement;

But it is only a matter of taste.